### PR TITLE
Support image size for generate image command

### PR DESCRIPTION
**Pull request by JeremyFabrikapp**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=e2NUf8l)
### Background

Discussion created here https://github.com/Torantulino/Auto-GPT/issues/492
One topic i would like to raise is the use of optional arguments, as Auto seems to ask for it anyway. Maybe we can add a kind of optional flag in the prompt.

### Changes

I have added an optional argument and fallback to default if not set.
image

### Test Plan
Use a prompt that will trigger the "generate_image" command and specify in natural language an image size.

### Change Safety
Only 3 sizes available at the moment, 256, 512, 1024 and with proper format, else fallback to 256x256.

- [x] I have added tests to cover my changes
- [x] I have considered potential risks and mitigations for my changes

### CODE REVIEW

**Comment by JeremyFabrikapp** [comment]
It's possible to continue adding params to improve image generation granularity, and adapting for specific SD image size.

**Comment by nponeccop** [comment]
@JeremyFabrikapp There are conflicts now

**Comment by JeremyFabrikapp** [comment]
Thanks @nponeccop conflicts have been fixed.

File: autogpt/commands/image_gen.py [hunk 1]
```diff
@@ -76,12 +81,13 @@ def generate_image_with_hf(prompt: str, filename: str) -> str:
...
 
 
-def generate_image_with_dalle(prompt: str, filename: str) -> str:
+def generate_image_with_dalle(prompt: str, filename: str, image_size: str) -> str:
     """Generate an image with DALL-E.
 
     Args:
         prompt (str): The prompt to use
         filename (str): The filename to save the image to
+        image_size (str): The image_size
```
Comment (kinance) [inline autogpt/commands/image_gen.py:90]: I think the parameter is missing size: int, please see the existing code is referring to size. Will be fixed in #2491.
```
    if size not in [256, 512, 1024]:
closest = min([256, 512, 1024], key=lambda x: abs(x - size))
```

Comment (JeremyFabrikapp) [reply autogpt/commands/image_gen.py:90]: Thanks @kinance I'll close the PR !

**Pull request closed without merging**